### DISCUSSION TITLE
JSII3001: Type "Map" cannot be used as the return type because it is private or @internal

### DISCUSSION THREAD

**Discussion by Microos** [aws/jsii#3849] [Q&A]
Why is Map not allowed in JSII? I got compile time error saying `JSII3001: Type "Map" cannot be used as the return type because it is private or @internal` for the following sample code

```typescript
// index.ts

export class Tester {
    constructor() {}

    test(): Map<string, number> {
        return new Map<string, number>([['12', 12]]);
    }
}
```

The error further refer to the constructor of map:
```
  node_modules/jsii/node_modules/typescript/lib/lib.es2015.collection.d.ts:36:13
    36 declare var Map: MapConstructor;
```

**Comment by bitfoo1** [comment]
Hi @Microos did you found a solution for that?

**Comment by Microos** [comment]
Hi, you simply cannot use Map while using JSII. Use the key value object as
map instead.
